### DISCUSSION TITLE
DHCP Server of Adguard cannot start in Version 4.4.0

### DISCUSSION THREAD

**Discussion by Fabstunes** [hassio-addons/addon-adguard-home#286] [General]
Hi!

I updated to 4.4.0 today but the DHCP Server was not able to start and so the add on was not stared either. I thought to create a issue but I am a realy new to github so I was not able to submit below. Please help me. I can of course check on a bugfix if this helps.

# Problem/Motivation
>Add On and DHCP Server could not start with fatal error: Cannot start on 172.16.0.1 - 172.16.0.255 with Gateway 172.16.0.1

## Expected behavior
>The DHCP Server should keep running and provide IP Addresses

## Actual behavior
>The Add On starts in a loop but never gets running

## Steps to reproduce
>Probably update to 4.4.0 and have the DHCP Server of Aduard in place

## Proposed changes
>

**Comment by AlecRust** [comment]
The `Gateway IP` must be _outside_ the `Range of IP addresses` have you limited the range to do this?

![image](https://user-images.githubusercontent.com/247634/147079534-046edc64-ac98-481f-b8cf-cd3b6f19be06.png)

**Reply by Fabstunes** [reply]
Thanks for the input. At the moment ist is working with
![image](https://user-images.githubusercontent.com/30566321/147095867-183d788d-44ea-442d-af92-b4351b3f9240.png)

I changed it to 
![image](https://user-images.githubusercontent.com/30566321/147096001-ce4ec68c-3d1d-4a0b-9060-235109d3a144.png)

and updated to 4.4.0
it works now :)
Thanks a lot

**Comment by ionuttimofte53** [comment]
Service keeps restarting... 

server...
2021/12/23 03:22:07.954859 [info] AdGuard Home, version v0.107.0
2021/12/23 03:22:07.960520 [fatal] initing dhcp: creating dhcpv4 srv: dhcpv4: gateway ip 10.128.191.191 in the ip range: 10.128.191.100-10.128.191.200

**Reply by AlecRust** [reply]
Yeah, your Gateway IP is not outside the range..

Change the range to not include the Gateway IP.

**Reply by ionuttimofte53** [reply]
Of course it's not outside. Ha is the DHCP server for my router. So my router is the gateway thru which all internet gets out ? Or am i getting it wrong? Before 2021.12.4, it functioned perfectly. Not sure why this sudden change. I can't access at all the Adguard page, to change any configuration inside it..... I can't change the gateway, even if i wanted to

**Reply by AlecRust** [reply]
> Of course it's not outside.

You seem to have not read the earlier comments in this discussion thread.. it's already been confirmed as a fix.

> Ha is the DHCP server for my router. So my router is the gateway thru which all internet gets out ? Or am i getting it wrong?

Correct.

> Before 2021.12.4, it functioned perfectly. Not sure why this sudden change.

I think it was more likely [v4.4.0](https://github.com/hassio-addons/addon-adguard-home/releases/tag/v4.4.0) of this add-on when this started happening. v4.4.0 includes a [major version bump](https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.0) of AdGuard Home.

> I can't access at all the Adguard page

Yeah you need to rollback to your backup of v4.3.0 of this add-on to get things working. Then change the `Range of IP addresses` to a range that does **not** include the `Gateway IP` (see screenshots etc. above). Then upgrade to v4.4.0 of this add-on.

`Range of IP addresses` is the range of IP addresses you want AdGuard Home DHCP server to assign to clients on your network. Your gateway (router) should have a fixed IP and should not be part of this range i.e. you do not want AdGuard Home to assign an IP address to it.
